### DISCUSSION TITLE
How to set test name at addon-interactions?

### DISCUSSION THREAD

**Discussion by soromii** [storybookjs/storybook#17804] [Help]
How to set test name at addon-interactions?
Is it available in beta version?
like this post. thanks

https://storybook.js.org/blog/interaction-testing-sneak-peek/

**Comment by TommasoAmici** [comment]
I'd be interested in this too, but looking at the source code of `@storybook/addons-interactions` this is not possible, I guess it was just a mock up?

**Comment by zyxneo** [comment]
From the documentation https://storybook.js.org/docs/essentials/interactions

Use `step` description.

```ts
// Form.stories.ts|tsx
...
export const Submitted: Story = {
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);
 
    await step('Enter credentials', async () => {
      await userEvent.type(canvas.getByTestId('email'), 'hi@example.com');
    });
 
    await step('Submit form', async () => {
      await userEvent.click(canvas.getByRole('button'));
    });
  },
};
```

**Reply by Stan-Stani** [reply]
https://storybook.js.org/docs/writing-tests/interaction-testing#group-interactions-with-the-step-function

**Comment by Stan-Stani** [comment]
Came here through google so going to leave this for others in 2025+ using Vitest: https://storybook.js.org/docs/writing-tests/integrations/vitest-addon#how-do-i-customize-a-test-name

> How do I customize a test name?
> 
> By default, the export name of a story is mapped to the test name. To create a more descriptive test description, you can provide a name property for the story. This allows you to include spaces, brackets, or other special characters.
> Example.stories.js|ts
> 
> export const Story = {
>   name: 'custom, descriptive name'
> };
